io_scene_gltf/node.py

```diff
@@ -0,0 +1,211 @@
+import bpy
+import math
+from mathutils import Matrix, Quaternion, Vector
+
+"""
+Handle nodes and scenes.
+
+The glTF node forest is represented in Blender as a single armature, where the
+nodes become bones. The meshes and cameras are set in the correct place in the
+heirachy using a "Copy Transform" constraint to lock them to their parent. Using
+an armature is necessary to skin meshes -- glTF skins through the configuration
+of the node forest and Blender only skins through armatures (AFAIK). The main
+drawback of this is that I don't think you can scale a bone's rest position so
+scaling nodes doesn't work. Also, the bone positions are fairly meaningless.
+
+COLLADA should have these problems too; check what it does? (I know it doesn't
+solve the bone positions issue.)
+
+For our purposes it would be really nice if Blender had an armature based on
+joints instead of bones.
+
+It would also be desirable to check that what we do gives the correst result ie.
+the transform Blender does equals the one glTF calls for, but neither Blender
+nor glTF have docs for exactly what that transform should be :-/
+
```

Comment:
I don't get the last paragraph. Can you explain?

Reply:
About scenes? If the glTF has nodes like this

       0       3
      / \     / \
     1   2   4   5
         |       |
       mesh1   mesh2

and there's a scene with root 0 and another with root 3, then we turn that into:

1. a Blender armature object A for all the nodes
2. two meshes, M1 for mesh1 and M2 for mesh 2, which we attach to A
3. for the first scene, we link in A and M1
4. for the second scene, we link in A and M2

Basically, for each scene, we link in A and then any object which is a descendant of one of the scene's roots.

In an older version I tried to partition it up so that there would be different armatures for the different scenes (eg. the first scene wouldn't contain node 3, 4, or 5) but the scenes can overlap and mesh1 could be skinned with eg. node 3, so I gave that up.

The glTF spec doesn't say much about what scenes actually do. The only prescriptive thing I could find was that the default scene should be "displayed at load time". Other than that a scene just marks some of the nodes in the forest as "roots".

I was also kind of considering just dropping scene handling too.